src/clientSideScene/sceneEntities.ts

```diff
@@ -798,18 +800,37 @@ export class SceneEntities {
             (sceneObject) => sceneObject.object.name === X_AXIS
           )
 
-          const lastSegment = sketch.paths.slice(-1)[0]
+          const lastSegment = sketch.paths.slice(-1)[0] || sketch.start
           const snappedPoint = {
             x: intersectsYAxis ? 0 : intersection2d.x,
             y: intersectsXAxis ? 0 : intersection2d.y,
           }
+          // Get the angle between the previous segment (or sketch start)'s end and this one's
+          const angle = Math.atan2(
+            snappedPoint.y - lastSegment.to[1],
+            snappedPoint.x - lastSegment.to[0]
+          )
+
+          const isHorizontal =
+            radToDeg(Math.abs(angle)) < ANGLE_SNAP_THRESHOLD_DEGREES ||
+            Math.abs(radToDeg(Math.abs(angle) - Math.PI)) <
+              ANGLE_SNAP_THRESHOLD_DEGREES
+          const isVertical =
+            Math.abs(radToDeg(Math.abs(angle) - Math.PI / 2)) <
+            ANGLE_SNAP_THRESHOLD_DEGREES
```

Comment:
Similar to how `isHorizontal` checks for 0 and 180 degrees, should this also check for 90 and 270 degrees? I.e. `3*PI / 2`.

Reply:
Oh yes but I found that `angle` is in the range of `(-180, 180]`, not `[0, 360)`, so I only need to check for `+90` and `-90` in practice.

Reply:
Here's a demo of snapping at what would normally be `90` and `270`

https://github.com/user-attachments/assets/9d6da0f3-c011-457f-967e-2a20a7040a33

Reply:
I realize that I reference `270º` in my description my bad, I still think in those terms